internal/fxreflect/fxreflect.go

```diff
@@ -27,77 +27,11 @@ import (
 	"regexp"
 	"runtime"
 	"strings"
-
-	"go.uber.org/dig"
 )
 
 // Match from beginning of the line until the first `vendor/` (non-greedy)
 var vendorRe = regexp.MustCompile("^.*?/vendor/")
 
-// ReturnTypes takes a func and returns a slice of string'd types.
-func ReturnTypes(t interface{}) []string {
-	if reflect.TypeOf(t).Kind() != reflect.Func {
-		// Invalid provide, will be logged as an error.
-		return []string{}
-	}
-
-	rtypes := []string{}
-	ft := reflect.ValueOf(t).Type()
-
-	for i := 0; i < ft.NumOut(); i++ {
-		t := ft.Out(i)
-
-		traverseOuts(key{t: t}, func(s string) {
-			rtypes = append(rtypes, s)
-		})
-	}
-
-	return rtypes
-}
-
-type key struct {
-	t    reflect.Type
-	name string
-}
-
-func (k *key) String() string {
-	if k.name != "" {
-		return fmt.Sprintf("%v:%s", k.t, k.name)
-	}
-	return k.t.String()
-}
-
-func traverseOuts(k key, f func(s string)) {
-	// skip errors
-	if isErr(k.t) {
```

Comment:
I think you can remove `isErr` since it's only being used here.

Reply:
Oh yeah, good catch. This should have been caught by staticcheck, but it looks like we don't have it set up for Fx. I'll follow up with that in a separate PR.